### ISSUE TITLE
trident genoconvert throws warning that files it created exist

### ISSUE THREAD

**Issue by TCLamnidis** [poseidon-framework/poseidon-hs#290]
I have created a package with eigenstrat genotypes. I then use `trident genoconvert` to convert the genotypes into PLINK format with the following command. 
```
package_name="2021_CarlhoffNature"
$(mktemp -d ~/.tmp/${package_name}.XXXXXXXXXX)

## Convert data to PLINK format
  trident genoconvert -d ${tmp_dir}/package \
    --genoFile ${tmp_dir}/package/${package_name}.geno \
    --snpFile ${tmp_dir}/package/${package_name}.snp \
    --indFile ${tmp_dir}/package/${package_name}.ind \
    --inFormat EIGENSTRAT \
    --outFormat PLINK \
    --removeOld
```
`tmp_dir` is a temp directory that was just created (`mktemp`) to ensure no other files interfere with anything, and contains a `package` subdirectory where the poseidon package resides.

This command returns the following log output:
```
[2021_CarlhoffNature] Converting data to PLINK format
trident v1.3.0.4 for poseidon v2.5.0, v2.6.0, v2.7.0, v2.7.1
https://poseidon-framework.github.io

[Info]    Checking base directories... 
[Info]    Searching POSEIDON.yml files... 
[Info]    1 found
[Info]    Checking Poseidon versions... 
[Info]    Initializing packages... 
[Info]    Packages loaded: 1
[Info]    Unpackaged genotype data files loaded: 1
[Info]    Converting genotype data in 2021_CarlhoffNature to format PLINK:
[Info]    Processing SNPs...
[Info]    SNPs:         0    0s
[Info]    Probing of the first 1000 SNPs successful. Continuing now...
[Info]    SNPs:     10000    0s
[...]
[Info]    SNPs:   1230000    26s
[Info]    Done
[Info]    Adjusting POSEIDON.yml...
[Info]    Converting genotype data in 2021_CarlhoffNature to format PLINK:
[Warning] File ~/.tmp/MNT_2021_CarlhoffNature.aTPeA4j1Al/package/2021_CarlhoffNature.bed exists
[Warning] File ~/.tmp/MNT_2021_CarlhoffNature.aTPeA4j1Al/package/2021_CarlhoffNature.bim exists
[Warning] File ~/.tmp/MNT_2021_CarlhoffNature.aTPeA4j1Al/package/2021_CarlhoffNature.fam exists
[Warning] skipping genotype conversion for 2021_CarlhoffNature
```

After converting the genotypes to the PLINK, `trident genoconvert` then seems to retry conversion before warning me that the files it just created exist. The line `[Info]    Converting genotype data in 2021_CarlhoffNature to format PLINK:` even comes up twice!

Interestingly, the PLINK files appear complete, and the POSEIDON.yml is correctly updated to the new PLINK data, when the command exits (with exit code 0, since these are warnings and not errors).

# trident version
```
$ trident --version
trident v1.3.0.4 for poseidon v2.5.0, v2.6.0, v2.7.0, v2.7.1
https://poseidon-framework.github.io

1.3.0.4
```

**Comment by stschiff** [comment]
Hmm, for starters, I think choosing both `-d` and the individual unpackaged genotype files with `--genoFile` etc. seems weird to me. Those options are meant to support converting unpackaged files, but you, in fact, have a package. So there is redundancy in your command, by providing both a base-path to a package plus the individual files. Perhaps we should make this clearer in the docs, but I think your command is Mal-formed.

I don't know whether there is something we can do to help with this... one option would be to change the input options in such a way that _either_ a single triple of unpackaged geno-files can be given, _or_ a base directory with packages in them.

The batch-option of genoconvert - I think - can come in handy to convert an entire set of packages, but I think once people use the unpackaged-genotype-files option, they don't want the batch-power, so we might as well restrict the unpackaged-behavior to be single-pac only. In other words, the API should support _either_ unpackaged triple _or_ base dir.

What do you think, @nevrome and @TCLamnidis?

**Comment by nevrome** [comment]
Very good spotting, Stephan!

I don't think that we should touch `GenoDataSource` or establish special rules in its reading just for `genoconvert`. Instead we could check if `properPackages` and `pseudoPackages` overlap regarding the source files and throw an exception if this is the case. For `forge` this could be a warning.

**Comment by stschiff** [comment]
Do you have comments, @TCLamnidis, on my diagnosis of this problem? Did you misunderstand how to use this tool, or would you like some additional safeguards as we discussed here?

**Comment by TCLamnidis** [comment]
I adjusted my command to this:
```
package_name="2021_CarlhoffNature"
$(mktemp -d ~/.tmp/${package_name}.XXXXXXXXXX)

  trident genoconvert -d ${tmp_dir}/package \
    --inFormat EIGENSTRAT \
    --outFormat PLINK \
    --removeOld
    ```
    
    and get the following error instead:
    ```
    trident v1.3.0.4 for poseidon v2.5.0, v2.6.0, v2.7.0, v2.7.1
https://poseidon-framework.github.io

Missing: (--genoFile FILE --snpFile FILE --indFile FILE)

Usage: trident genoconvert ((-d|--baseDir DIR) | 
                             ((-p|--genoOne FILE) | --inFormat FORMAT
                               --genoFile FILE --snpFile FILE --indFile FILE) 
                             [--snpSet SET]) --outFormat FORMAT [--onlyGeno] 
                           [-o|--outPackagePath DIR] [--removeOld] 
                           [--outPlinkPopName MODE]

  Convert the genotype data in a Poseidon package to a different file format
  ```
  
  So it seems that just using `-d` doesn't work, as I thought it would from your comment. Using the `--*File` options seems to work though

**Comment by TCLamnidis** [comment]
Oh, It seems adding `--inFormat` automatically disregards the `-d` and expects `--*File` options to appear. removing that worked.
```
  trident genoconvert \
    -d ${tmp_dir}/package \
    --outFormat PLINK \
    --removeOld
```

**Comment by TCLamnidis** [comment]
To answer @stschiff question. In hindsight, the usage string is pretty clear about the grouping of all the parameters and what is needed. will close this issue.

**Comment by stschiff** [comment]
Yeah, the specific library we're using to parse command line is pretty sophisticated, and I admit the complexity sometimes goes slightly beyond what people are used to 😅
